### PR TITLE
Add timer.get_info() function

### LINKED ISSUE #6995: Timer - get time remaining until next call
Would like a function to get the time remaining until the next time a timer.delay() fires.

Example use case: I fire player weapons using timer.delay(). The player obtains a weapon cooldown upgrade, which means I need to restart all weapon timers in a new interval. However, I do not know how near a shot the weapon timers are, so either have to reset them entirely or implement a workaround.

### PR DESCRIPTION
**Pull request by d954mas**
Sometimes it is needed to get additional information about a running timer, for instance the remaining time before it fires. This change adds a `timer.get_info(handle)` method which returns a table with information about a timer. Example:

```
local handle = timer.delay(2, true, function() end)

--after 0.5 seconds we call:

local info = timer.get_info(handle)
print(info.time_remaining) -- 1.5
print(info.delay)     -- 2
print(info.repeating) -- true
```

If timer cancelled or completed the function will return nil.

Fixes #6995

### CODE REVIEW

File: engine/script/src/script_timer.cpp [hunk 1]
```diff
@@ -676,10 +676,69 @@ namespace dmScript
         return 1;
     }
 
+    /*# get information about timer
+     *
+     * Get information about timer.
+     *
+     * @name  timer.get_info
+     * @param handle [type:hash] the timer handle returned by timer.delay()
+     * @return data [type:table] or nil if timer is cancelled/completed. table with data in the following fields:
```
Comment (d954mas) [inline engine/script/src/script_timer.cpp:685]: Return table or nil.
How to mark it correctly?

**Comment by britzl** [comment]
@d954mas FYI I updated the initial PR description. We use the first post of the PR when we generate release notes.

**Comment by d954mas** [comment]
ping:)

**Comment by britzl** [comment]
> ping:)

Sorry! I'll try to get it into the new release!

**Review by britzl — APPROVED** [review]

**Pull request merged**